src/main/java/frc/robot/subsystems/LightStrip.java

```diff
@@ -30,17 +30,22 @@ public void periodic() {
                 .ifPresent(c -> {
                     setPattern(predicateLightEvents.get(c));
                     predicateLightEvents.remove(c);
-                    Logger.logInteger(baseLogName + "predicateSize", predicateLightEvents.size(), Constants.ENABLE_LOGGING);
+                    if (Constants.LED_DEBUG){
+                        Logger.logInteger(baseLogName + "predicateSize", predicateLightEvents.size(), Constants.ENABLE_LOGGING);
+                    }
                 }
         );
     }
 
     public void setPattern(BlinkinPattern pattern) {
         this.pattern = pattern;
         colorSensorPort.set(pattern.getPwm());
-        SmartDashboard.putString("PATTERN",pattern.toString());
-        Logger.logDouble(baseLogName + "pwmSignal", pattern.getPwm(), Constants.ENABLE_LOGGING);
-        Logger.logString(baseLogName + "pwmName", pattern.toString(), Constants.ENABLE_LOGGING);
+        if (Constants.LED_DEBUG){
+            SmartDashboard.putString("PATTERN",pattern.toString());
+            Logger.logDouble(baseLogName + "pwmSignal", pattern.getPwm(), Constants.ENABLE_LOGGING);
```

Comment:
I would vote for logging the signal whenever it's set, were these really that expensive? (leaving SmartDashboard  as debug makes sense)